exchanges/deribit/deribit.go

```diff
@@ -0,0 +1,2780 @@
+func (d *Deribit) GetSupportedIndexNames(ctx context.Context, priceIndexType string) (interface{}, error) {
+	params := url.Values{}
+	if priceIndexType != "" {
+		params.Set("type", priceIndexType)
+	}
+	var resp []string
+	return resp, d.SendHTTPRequest(ctx, exchange.RestSpot, nonMatchingEPL, common.EncodeURLValues("public/get_supported_index_names", params), &resp)
+}
+
+// GetRequestForQuote retrieves RFQ information.
+func (d *Deribit) GetRequestForQuote(ctx context.Context, ccy, kind string) ([]RequestForQuote, error) {
+	if ccy == "" {
+		return nil, fmt.Errorf("%w '%s'", errInvalidCurrency, ccy)
+	}
+	params := url.Values{}
+	params.Set("currency", ccy)
+	if kind != "" {
+		params.Set("kind", kind)
+	}
+	var resp []RequestForQuote
+	return resp, d.SendHTTPRequest(ctx, exchange.RestFutures, nonMatchingEPL, common.EncodeURLValues(getRFQ, params), &resp)
+}
+
+// GetTradeVolumes gets trade volumes' data of all instruments
+func (d *Deribit) GetTradeVolumes(ctx context.Context, extended bool) ([]TradeVolumesData, error) {
+	params := url.Values{}
+	if extended {
+		params.Set("extended", "true")
+	}
+	var resp []TradeVolumesData
+	return resp, d.SendHTTPRequest(ctx, exchange.RestFutures, nonMatchingEPL,
+		common.EncodeURLValues(getTradeVolumes, params), &resp)
+}
+
+// GetTradingViewChart gets volatility index data for the requested instrument
+func (d *Deribit) GetTradingViewChart(ctx context.Context, instrument, resolution string, startTime, endTime time.Time) (*TVChartData, error) {
+	params, err := checkInstrument(instrument)
+	if err != nil {
+		return nil, err
+	}
+	err = common.StartEndTimeCheck(startTime, endTime)
+	if err != nil {
+		return nil, err
+	}
+	if resolution == "" {
+		return nil, errResolutionNotSet
+	}
+	params.Set("start_timestamp", strconv.FormatInt(startTime.UnixMilli(), 10))
+	params.Set("end_timestamp", strconv.FormatInt(endTime.UnixMilli(), 10))
+	params.Set("resolution", resolution)
+	var resp *TVChartData
+	return resp, d.SendHTTPRequest(ctx, exchange.RestFutures, nonMatchingEPL,
+		common.EncodeURLValues(getTradingViewChartData, params), &resp)
+}
+
+// GetResolutionFromInterval returns the string representation of intervals given kline.Interval instance.
+func (d *Deribit) GetResolutionFromInterval(interval kline.Interval) (string, error) {
+	switch interval {
+	case kline.HundredMilliseconds:
+		return "100ms", nil
+	case kline.OneMin:
+		return "1", nil
+	case kline.ThreeMin:
+		return "3", nil
+	case kline.FiveMin:
+		return "5", nil
+	case kline.TenMin:
+		return "10", nil
+	case kline.FifteenMin:
+		return "15", nil
+	case kline.ThirtyMin:
+		return "30", nil
+	case kline.OneHour:
+		return "60", nil
+	case kline.TwoHour:
+		return "120", nil
+	case kline.ThreeHour:
+		return "180", nil
+	case kline.SixHour:
+		return "360", nil
+	case kline.TwelveHour:
+		return "720", nil
+	case kline.OneDay:
+		return "1D", nil
+	case kline.Interval(0):
+		return "raw", nil
```

Comment:
Maybe make an exported  kline package Raw set to `-1` so its a specific time assignment that we can check for and its actually requested.